From 6c0502df0016773226744999766f976b110378ea Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Fri, 31 Oct 2025 21:31:31 +0100
Subject: [PATCH] Set colors for focused item in the main menu

This makes the keyboard navigation more usable as the actually focused item
is highlighted.
---
 src/theme/common/_menu.scss | 29 ++++++++++++++++++++++++++++-
 1 file changed, 28 insertions(+), 1 deletion(-)

diff --git a/src/theme/common/_menu.scss b/src/theme/common/_menu.scss
index 0411649e0..2ccaf279f 100644
--- a/src/theme/common/_menu.scss
+++ b/src/theme/common/_menu.scss
@@ -48,6 +48,14 @@
         > row {
             padding: 0;
             margin: 0;
+
+            &:focus {
+                image, label {
+                    color: $selected_fg_color;
+                }
+                background-color: $selected_bg_color;
+            }
+
         }
 
         button.flat {
@@ -96,6 +104,12 @@
 
             &:hover { background-color: transparentize(white, 0.9); }
             &:active { background-color: transparentize(white, 0.8); }
+            &:focus {
+                image, label {
+                    color: $selected_fg_color;
+                }
+                background-color: $selected_bg_color;
+            }
         }
 
         .user-icon-button {
@@ -108,6 +122,12 @@
 
             &:hover { background-color: transparentize(white, 0.9); }
             &:active { background-color: transparentize(white, 0.8); }
+            &:focus {
+                image, label {
+                    color: $selected_fg_color;
+                }
+                background-color: $selected_bg_color;
+            }
         }
     }
 
@@ -119,6 +139,13 @@
         margin-bottom: 6px;
         padding: 8px;
 
+        > row:focus {
+            image, label {
+                color: $selected_fg_color;
+            }
+            background-color: $selected_bg_color;
+        }
+
         button {
             @extend %reset_style;
 
@@ -160,7 +187,7 @@
         transition: 170ms ease-out;
 
         &:hover { background-color: transparentize(white, 0.9); }
-        &:active, &:checked {
+        &:active, &:checked, &:focus {
             label {
                 color: $selected_fg_color;
             }
